public/js/Histogram.js

```diff
@@ -0,0 +1,95 @@
+var d3 = require('d3');
+var _ = require('lodash');
+_.mixin(require('lodash-deep'));
```

Comment:
``lodash-deep`` adds a few methods to lodash that support operating on nested properties of objects. It adds a few different methods.. the 'circle slash circle' function ``%``, 'low-res snowflake' ``*`` and 'tic-tac-toe' ``#`` :P Kidding.. they're called from lodash, so they're just called from ``_``. See line 25:

``
var featureDist = _.deepPluck(self.sampleData,
        ['feature_vector_std', feature]);
``

This is reading the value of index at ``feature`` in the property ``feature_vector_std`` for every object in the ``sampleData`` array. It can access properties nested even further too. I was originally using my own function to do this, but I prefer this method. Say what you like about their naming conventions, the lodash devs are optimisation obsessed so I'll use their methods whenever appropriate!